IPython/core/magics/code.py

```diff
@@ -656,6 +660,7 @@ def edit(self, parameter_s='',last_call=['','']):
             args = str(e.index)
             filename, lineno, is_temp = self._find_edit_target(self.shell, 
                                                        args, opts, last_call)
+            self._knowntemps.add(filename)
```

Comment:
I think this should be outside the `except`, and depend on the `is_temp` variable.